src/hooks/useDragThoughtAnimation.ts

```diff
@@ -0,0 +1,59 @@
+import { throttle } from 'lodash'
+import { useLayoutEffect, useMemo, useState } from 'react'
+import { useSelector } from 'react-redux'
+import State from '../@types/State'
+import durations from '../durations.config'
+import usePrevious from './usePrevious'
+
+/**
+ * Returns a style that applies the "moveThoughtOver" animation
+ * when this thought was part of the most recent drag-and-drop move.
+ */
+const useDragThoughtAnimation = (index: number, isDragged: boolean): React.CSSProperties | undefined => {
+  // Detect if the last patch included a drag-and-drop moveThought.
+  // Aggregate actions across the whole patch to handle merged multicursor operations.
+  const isLastActionMoveThought = useSelector((state: State): boolean => {
+    const lastPatches = state.undoPatches[state.undoPatches.length - 1]
+    const actionsInPatch = new Set(
+      (lastPatches ?? []).flatMap(op => (op as unknown as { actions?: string[] }).actions || []),
+    )
+    return actionsInPatch.has('moveThought')
```

Comment:
Instantiating a Set is unnecessary if we're just checking if any operation in the patch contains `moveThought`. Maybe you use a nested `some`?

Reply:
I've updated it with `some`.